timelock-agent/src/main/java/com/palantir/timelock/paxos/AsyncTimeLockServicesCreator.java

```diff
@@ -84,6 +88,8 @@ public TimeLockServices createTimeLockServices(
                         : JavaSuppliers.compose(NonTransactionalLockService::new, rawLockServiceSupplier),
                 client);
 
+        leadershipCreator.executeWhenLostLeadership(this::deregisterLeaderMetrics);
```

Comment:
Do we need a similar method when we gain leadership to deregister any metrics with `isCurrentSuspectedLeader=false`?

Reply:
It could be useful. I don't think we're likely to watch the `false` metrics as closely (e.g. we're unlikely to use them for monitors), but it would be good to eliminate potentially confusing information.